src/context/WalletProvider/KeepKeyProvider.tsx

```diff
@@ -130,6 +151,71 @@ export const KeepKeyProvider = ({ children }: { children: React.ReactNode }): JS
     })()
   }, [keepKeyWallet, keepKeyWallet?.features, setDeviceTimeout, setHasPassphrase])
 
+  useEffect(() => {
+    if (!keepKeyWallet) return
+    ;(async () => {
+      const features = await keepKeyWallet.getFeatures()
+      const versions = await getKeepKeyVersions(keepKeyWallet, features?.bootloaderHash)
+
+      if (!versions) return
+
+      if (
+        (versions.bootloader.updateAvailable || versions.firmware.updateAvailable) &&
+        !toast.isActive(KeepKeyToastId)
+      ) {
+        toastRef.current = toast({
+          render: () => {
```

Comment:
@0xApotheosis I'm not a big fan about this:

The render function doesn't have access to the `react-polygot` provider, meaning that the component can't use the `useTranslate` hook, this means that we are forced to add every already translated strings as a prop including the beginning, end and center of the description sentences

As a quickfix, I passed the translate function as a prop but it feels really dirty to me, did I miss something?

Reply:
`UpdateToastAvailable.tsx` is a React component, so we can use the translation provider directly. 

Just remove the arg and add `const translate = useTranslate()` in the file.

Decouple your `onClose()` handler into a callback:

```ts
const onClose = useCallback(() => {
    if (toastRef.current) {
      toast.close(toastRef.current)
    }
  }, [toast])
```

And then then all of this simply becomes:

```ts
toastRef.current = toast({
  render: () => <UpdateAvailableToast onClose={onClose} />,
  id: KeepKeyToastId,
  duration: null,
  isClosable: true,
  position: 'bottom-right',
})
```

Reply:
We can't, probably because `<UpdateAvailableToast />` is inside a render function, which is probably not inside the translation provide, this means that `useTranslate` return undefined everytime

As said in private messages, I'll just restore the markup without any decoupled component

Reply:
@NeOMakinG Can you confirm that decoupling and trying to use `useTranslate()` doesn't work and `useTranslate` returns undefined? That's an odd one, it should be drilled down from the context provider, but if it doesn't work, happy with keeping the JSX node here.

Reply:
Worst case scenario we could pass down `translate` as a prop which will remember its context (yay for closures!)